__tests__/ProfileImageList-test.js

```diff
@@ -0,0 +1,161 @@
+  it('Profile View state = grid and 3 column flatlist is present', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    const instance = wrapper.instance();
+    instance.setState({
+      profile_view: 'grid',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- for what reason? */
+    //const viewState = instance.state.profile_view;
+    //expect(viewState).toEqual('grid');
+
+    const flatlistWrapper = wrapper.find(FlatList);
+    //const threeColWrapper = flatlistWrapper.find({numColumns: 3});
+    //console.log(threeColWrapper.first().props().numColumns);
+
+    expect(flatlistWrapper.find({numColumns: 3})).toHaveLength(1);
+  });
+
+  it('Profile View state = grid and 2 column flatlist is NOT present', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    const instance = wrapper.instance();
+    instance.setState({
+      profile_view: 'grid',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- for what reason? */
+    //const viewState = instance.state.profile_view;
+    //expect(viewState).toEqual('grid');
+
+    const flatlistWrapper = wrapper.find(FlatList);
+    //const threeColWrapper = flatlistWrapper.find({numColumns: 3});
+    //console.log(threeColWrapper.first().props().numColumns);
+
+    expect(flatlistWrapper.find({numColumns: 1})).toHaveLength(0);
+  });
+
+  it('Profile View state = list and 1 column flatlist is present', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    const instance = wrapper.instance();
+    instance.setState({
+      profile_view: 'list',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- For what reason? */
+    //const viewState = instance.state.profile_view;
+    //expect(viewState).toEqual('grid');
+
+    const flatlistWrapper = wrapper.find(FlatList);
+    //const threeColWrapper = flatlistWrapper.find({numColumns: 3});
+    //console.log(threeColWrapper.first().props().numColumns);
+
+    expect(flatlistWrapper.find({numColumns: 1})).toHaveLength(1);
+  });
+
+  it('Profile View state = list and 3 column flatlist is NOT present', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    const instance = wrapper.instance();
+    instance.setState({
+      profile_view: 'list',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- For what reason?*/
+    //const viewState = instance.state.profile_view;
+
+    //expect(viewState).toEqual('grid');
+
+    const flatlistWrapper = wrapper.find(FlatList);
+    //const threeColWrapper = flatlistWrapper.find({numColumns: 3});
+    //console.log(threeColWrapper.first().props().numColumns);
+
+    expect(flatlistWrapper.find({numColumns: 3})).toHaveLength(0);
+  });
+}
+
+testWhichFlatlistShows();
+
+// function testOnViewHeadlineIconButtonClicked(initalProfileViewStatus){
```

Comment:
Consider deleting unnecessary comments. Previous code can be recovered from past commits